### ISSUE TITLE
SIGHUP kills vtgate if no auth is configured

### ISSUE THREAD

**Issue by gedgar** [vitessio/vitess#6290]
#### Overview of the Issue
While you can send vtgate a `SIGHUP` to reload it's `mysql_auth_server_static_file` JSON credentials file, I noticed during some testing that a `SIGHUP` will actually kill vtgate entirely if `mysql_auth_server_impl` is set to `none` or if both `mysql_auth_server_static_file` and `mysql_auth_server_static_string` are empty. 

#### Reproduction Steps
1. Launch the local example
2. Confirm vtgate is running w/ `mysql_auth_server_impl none`:
```
ps aux | grep [v]tgate
gary     17067  0.1  0.3 1940404 43372 pts/2   Rl   11:21   0:00 vtgate -topo_implementation etcd2 -topo_global_server_address localhost:2379 -topo_global_root /vitess/global -log_dir /home/gary/vtdataroot/tmp -log_queries_to_file /home/gary/vtdataroot/tmp/vtgate_querylog.txt -port 15001 -grpc_port 15991 -mysql_server_port 15306 -mysql_server_socket_path /tmp/mysql.sock -cell zone1 -cells_to_watch zone1 -tablet_types_to_wait MASTER,REPLICA -gateway_implementation discoverygateway -service_map grpc-vtgateservice -pid_file /home/gary/vtdataroot/tmp/vtgate.pid -mysql_auth_server_impl none
```
3. Send SIGHUP
`kill -1 $(pgrep vtgate)`
4. Note that vtgate has been killed and is no longer running. 

#### Binary version
`Version: 945e105aa (Git branch 'master') built on Tue Jun  9 11:20:38 PDT 2020 by gary@xj13 using go1.14 linux/amd64`

#### Log Fragments
I didn't notice anything relevant/useful in Vitess' logs.
strace of vtgate being killed by via `SIGHUP`: 
```
--- SIGHUP {si_signo=SIGHUP, si_code=SI_USER, si_pid=13723, si_uid=1000} ---
rt_sigprocmask(SIG_UNBLOCK, [HUP], NULL, 8) = 0
getpid()                                = 17067
gettid()                                = 17067
tgkill(17067, 17067, SIGHUP)            = 0
--- SIGHUP {si_signo=SIGHUP, si_code=SI_TKILL, si_pid=17067, si_uid=1000} ---
rt_sigaction(SIGHUP, {sa_handler=SIG_DFL, sa_mask=~[], sa_flags=SA_RESTORER|SA_ONSTACK|SA_RESTART|SA_SIGINFO, sa_restorer=0x7f4222daa890}, NULL, 8) = 0
rt_sigprocmask(SIG_UNBLOCK, [HUP], NULL, 8) = 0
getpid()                                = 17067
gettid()                                = 17067
tgkill(17067, 17067, SIGHUP)            = 0
--- SIGHUP {si_signo=SIGHUP, si_code=SI_TKILL, si_pid=17067, si_uid=1000} ---
+++ killed by SIGHUP +++
```

**Comment by buraksezer** [comment]
@harshit-gangal I would like to take this issue.

**Comment by harshit-gangal** [comment]
@buraksezer Go ahead :)

**Comment by sahutd** [comment]
@harshit-gangal  given that the issue was marked as a "good first issue", i was trying to fix it. but it seems to be fixed since `14ab70327e5da456dfb6bf03b1dad143044e64a4`.  [ran git bisect between `945e105aa` reported by @gedgar and latest commit in main branch. can you confirm this?

**Comment by harshit-gangal** [comment]
This is not a problem on main.
@sahutd Thank you for trying it out.

@gedgar Resolving it as it cannot be reproduced.
